pelican/tools/pelican_import.py

```diff
@@ -667,7 +667,12 @@ def download_attachments(output_path, urls):
             os.makedirs(full_path)
         print('downloading {}'.format(filename))
         try:
-            urlretrieve(url, os.path.join(full_path, filename))
+            if(url in successful_url_cache):
+                print('already downloaded: {} ... skipping'.format(filename))
+            else:
+                quote_url = quote(url, safe="%/:=&?~#+!$,;'@()*[]")
+                urlretrieve(quote_url, os.path.join(full_path, filename))
+                successful_url_cache.add(url)
```

Comment:
Does every call to `urlretrieve` generate a successful url?
Is there a way to test if `urlretrieve` was actually successful and only then add it to `successful_url_cache`?